### PR TITLE
lib: Add support for action validate-all-xml

### PR DESCRIPTION
**Pull request by marxsk**
Output of this action is machine-readable format in the XML format that
can be used for processing in pacemaker/pcs/UI

### CODE REVIEW

**Comment by marxsk** [comment]
@oalbrigt 
It will be required to update XML metadata as well but I did not want to enlarge PR yet. 

Please, take a look at code and possible extensions also from the point of resource-agents

File: lib/fencing.py.py [hunk 1]
```diff
@@ -711,10 +711,16 @@ def check_input(device_opt, opt, other_conditions = False):
 		options["--action"] = "off"
 
 
-	if options["--action"] == "validate-all" and not other_conditions:
-		if not _validate_input(options, False):
```
Comment (oalbrigt) [inline lib/fencing.py.py]: I think it'd be better to make a variable called e.g. validate_format to use here, so we can skip the elif part below.

File: lib/fencing.py.py [hunk 2]
```diff
@@ -1295,64 +1301,130 @@ def _set_default_values(options):
...
+		})
 
 	if device_opt.count("ipaddr") and "--ip" not in options and "--managed" not in options and "--target" not in options:
-		valid_input = False
-		fail_usage("Failed: You have to enter fence address", stop)
+		errors.append({
+			'fields': ['ipaddr', 'managed', 'target'],
+			'text': 'You have to enter fence address',
+			'text_cli': 'You have to enter fence address (-a)',
+			'error_type': 'REQUIRE',
```
Comment (oalbrigt) [inline lib/fencing.py.py]: Is this one-or-more, vs one-of being one or the other?

File: lib/fencing.py.py [hunk 3]
```diff
@@ -711,10 +711,16 @@ def check_input(device_opt, opt, other_conditions = False):
...
 
 
-	if options["--action"] == "validate-all" and not other_conditions:
-		if not _validate_input(options, False):
-			fail_usage("validate-all failed")
-		sys.exit(EC_OK)
+	if options["--action"] in ["validate-all", "validate-all-xml"] and not other_conditions:
+		if options["--action"] == "validate-all":
+			if not _validate_input(options, False, 'text'):
+				fail_usage("validate-all failed")
```
Comment (marxsk) [inline lib/fencing.py.py]: add sys.exit(EC_GENERIC_ERROR)

File: lib/fencing.py.py (hunk 1 above)
Comment (marxsk) [reply lib/fencing.py.py]: I like the idea of having validate_actions. But it won't help us with if/elif part, the code still have to be branched.

File: lib/fencing.py.py (hunk 2 above)
Comment (marxsk) [reply lib/fencing.py.py]: one-or-more; which name do you suggest? ONE-OR-MORE?

**Comment by oalbrigt** [comment]
add to whitelist

**Comment by oalbrigt** [comment]
add to whitelist

**Comment by oalbrigt** [comment]
retest this please

**Comment by fabbione** [comment]
@marxsk  apparently github has closed this issue when renaming master branch to main. I am unable to reopen the PR for you unfortunately, as you need to change the target branch from master to main.

**Pull request closed without merging**